helpers/location_service.py

```diff
@@ -1,166 +1,240 @@
-import json
-import ssl
-import threading
-import time
+from json import dumps, loads
```

Comment:
I'm not perfect about it, but best practice as I'm aware is to import modules, not classes/functions directly (though obviously QT has a different pattern, and they encourage class imports). Granted this is a style choice so I'm not going to harp on it too hard, but if possible would you mind leaving module imports where possible? Unless there's a counter-argument to this I haven't heard yet, which is also possible :P